ArduPlane/GCS_Mavlink.cpp

```diff
@@ -974,8 +974,7 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_long_packet(const mavlink_command_l
         }
 
         AP_Mission::Mission_Command cmd;
-        if (AP_Mission::mavlink_cmd_long_to_mission_cmd(packet, cmd) == MAV_MISSION_ACCEPTED) {
-            plane.do_change_speed(cmd);
+        if ((AP_Mission::mavlink_cmd_long_to_mission_cmd(packet, cmd) == MAV_MISSION_ACCEPTED) && plane.do_change_speed(cmd)) {
```

Comment:
Sorry, this isn't right.

One indicates that the packet was malformed (`MAV_RESULT_DENIED`), the other is we couldn't do what was demanded `MAV_RESULT_FAILED`)

Reply:
Sorry, I'm struggling to follow. You mean for the existing call to `AP_Mission::mavlink_cmd_long_to_mission_cmd(packet, cmd)` ?

Reply:
fixed, thanks